Quotient/connection.cpp

```diff
@@ -320,13 +320,15 @@ void Connection::Private::completeSetup(const QString& mxId, bool mock)
     }
 
     if (useEncryption) {
-        if (auto&& maybeEncryptionData =
-                _impl::ConnectionEncryptionData::setup(q, mock)) {
-            encryptionData = std::move(*maybeEncryptionData);
-        } else {
-            useEncryption = false;
-            emit q->encryptionChanged(false);
-        }
+        _impl::ConnectionEncryptionData::setup(q, [this](auto&& maybeEncryptionData){
+            if (maybeEncryptionData) {
+                encryptionData = std::move(*maybeEncryptionData);
+            } else {
+                useEncryption = false;
+                emit q->encryptionChanged(false);
+            }
+            emit q->ready();
```

Comment:
Right now, `TestOlmAccount` fails on my machine because `ready()` comes later than `connected()`, and the test uses `connected()` to proceed with its E2EE stuff, which is not there yet. I like the new signal name but I think having both signals together would confuse client authors. Should we keep `Connection::ready()` as the only recommended signal for clients, and deprecate `connected()` but fire it next to `ready()` in the meantime?

Reply:
yes, makes sense